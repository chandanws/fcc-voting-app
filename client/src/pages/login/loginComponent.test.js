import React from "react";
import LoginComponent from "./loginComponent";
import { shallow } from "enzyme";

describe("<HomepageComponent />", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<LoginComponent />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calling handleChange should change state", () => {
    expect(wrapper.state().username).toBe("");
    expect(wrapper.state().password).toBe("");
    wrapper.instance().handleChange(1)({ target: { value: "username" } });
    wrapper.instance().handleChange(2)({ target: { value: "pass" } });
    wrapper.update();
    expect(wrapper.state().username).toBe("username");
    expect(wrapper.state().password).toBe("pass");
  });
});
