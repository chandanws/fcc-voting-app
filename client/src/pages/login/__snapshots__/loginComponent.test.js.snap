// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomepageComponent /> should render correctly 1`] = `
<div>
  <Typography
    component="h1"
    text="Login"
  />
  <Form
    onSubmit={[Function]}
  >
    <Component
      component="input"
      id={1}
      label="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <Component
      component="input"
      id={2}
      label="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </Form>
</div>
`;
